framework/src/modules/chain/blocks/chain.js

```diff
@@ -195,11 +224,29 @@ const saveBlock = async (storage, block, tx) => {
 	parsedBlock.previousBlockId = parsedBlock.previousBlock;
 	delete parsedBlock.previousBlock;
 
-	parsedBlock.transactions.map(transaction => {
+	parsedBlock.transactions.forEach(transaction => {
```

Comment:
Avoid using forEach if possible.

Reply:
What do you suggest?

Reply:
`for (transaction of parsedBlock.transactions)`. However we haven't agreed about this yet so feel free to adopt it if the scope of the PR doesn't get too big.

Reply:
I posted earlier to Slack about this topic but it was not concluded. Let's conclude the topic outside PR and then all follow same pattern.